src/main/java/com/zendesk/maxwell/schema/ddl/TableAlter.java

```diff
@@ -34,29 +31,17 @@ public String toString() {
 	}
 
 	@Override
-	public Schema apply(Schema originalSchema) throws SchemaSyncError {
-		Schema newSchema = originalSchema.copy();
-
-		Database database = newSchema.findDatabase(this.database);
-		if ( database == null ) {
-			throw new SchemaSyncError("Couldn't find database: " + this.database);
-		}
-
-		Table table = database.findTable(this.table);
-		if ( table == null ) {
-			throw new SchemaSyncError("Couldn't find table: " + this.database + "." + this.table);
-		}
+	public ResolvedTableAlter resolve(Schema schema) throws SchemaSyncError {
+		Database database = schema.findDatabaseOrThrow(this.database);
+		Table oldTable = database.findTableOrThrow(this.table);
 
+		Table table = oldTable.copy();
 
 		if ( newTableName != null && newDatabase != null ) {
-			Database destDB = newSchema.findDatabase(this.newDatabase);
-			if ( destDB == null )
-				throw new SchemaSyncError("Couldn't find database " + this.database);
-
-			table.rename(newTableName);
+			schema.findDatabaseOrThrow(this.newDatabase);
 
-			database.getTableList().remove(table);
-			destDB.addTable(table);
+			table.name = newTableName;
```

Comment:
ah, good call.  I will add some tests around renaming tables on a lower-case server.  I think I can do that...